src/declaration.d

```diff
@@ -1958,18 +1959,38 @@ public:
             printf("%llx, isModule: %p, isTemplateInstance: %p\n", storage_class & (STCstatic | STCconst), parent.isModule(), parent.isTemplateInstance());
             printf("parent = '%s'\n", parent.toChars());
         }
-        if (!canTakeAddressOf())
-            return false;
-        Dsymbol parent = toParent();
-        if (!parent && !(storage_class & STCstatic))
+                
+        if (isdataseg == 0) // the value is not cached
         {
-            error("forward referenced");
-            type = Type.terror;
-            return false;
+
+            if (!canTakeAddressOf())
+            {
+                isdataseg = 0b11;
+                return false;
+            }
+
+            Dsymbol parent = toParent();
+            if (!parent && !(storage_class & STCstatic))
+            {
+                error("forward referenced");
+                type = Type.terror;
+                isdataseg = 0b11;
```

Comment:
Since the value is basically a 2bit bit-field I thought it appropriate to write it as binary literal.
Otherwise it could be difficult to see why the value 2 is not used;
0 // unset
1 // true
3 // false